### PR TITLE
Backward compatibility with py2 for TSV<->CSV conversion

### LINKED ISSUE #9341: TSV<->CSV conversion failing
```
Traceback (most recent call last):
  File "/home/dave/github-repos/galaxy/lib/galaxy/datatypes/converters/tabular_csv.py", line 44, in <module>
    main()
  File "/home/dave/github-repos/galaxy/lib/galaxy/datatypes/converters/tabular_csv.py", line 22, in main
    convert_to_csv(input_fname, output_fname)
  File "/home/dave/github-repos/galaxy/lib/galaxy/datatypes/converters/tabular_csv.py", line 40, in convert_to_csv
    writer.writerow(line.strip().split('\t'))
TypeError: a bytes-like object is required, not 'str'
```

### PR DESCRIPTION
**Pull request by davebx**
Resolves #9341

### CODE REVIEW

File: lib/galaxy/datatypes/converters/tabular_csv.py
Comment (jdavcs) [inline lib/galaxy/datatypes/converters/tabular_csv.py]:
```suggestion
    try:
        yield fh
    finally:
        fh.close()
```
The file needs to be closed upon exit from the context manager; I also think the yield statement needs to be wrapped in a try/finally: if an exception is raised in the `with` block, it will be re-raised in the `yield` statement. Without error handling whatever follows the yield statement (`fh.close()`) won't be executed.

I'm not sure whether we should worry about not wrapping the code that actually opens the file. Wrapping it in the same try block seems wrong, but multiple try blocks seem like overkill here. Depends on what we want the system to do if the file doesn't open here; and I don't know.

**Review by jdavcs — CHANGES_REQUESTED** [review]
Nice solution! (see inline comment).
Other than that, just 3 minor lint errors.

**Comment by nsoranzo** [comment]
This can be simplified by using only the Py3 syntax if https://github.com/galaxyproject/galaxy/pull/9345/files#diff-810d43c5eb8bd1186917f323691dd578 gets merged.

File: lib/galaxy/datatypes/converters/tabular_csv.py
Comment (jdavcs) [reply lib/galaxy/datatypes/converters/tabular_csv.py]: Correction: see comment by @nsoranzo

**Comment by jdavcs** [comment]
1. The unit test failure is unrelated.
2. I'm ambivalent about merging as is vs removing py2 compatibility altogether, pending #9345 as per @nsoranzo . On the one hand, the current version fixes a problem that currently exists in a reasonable way, without breaking the converter under Py2. On the other hand, once 9345 is merged, this will have introduced more "cruft" into the code base. 
I'll go ahead and approve this. Unless others feel it's better to wait for 9345, my suggestion is to merge this as is.

**Review by jdavcs — APPROVED** [review]
See previous comment about merging as is vs. waiting for the referenced PR.

**Comment by jdavcs** [comment]
#9345 has been merged, so we don't need to make this Python2-compatible anymore.

**Comment by mvdbeek** [comment]
Thanks @davebx, I'm gonna close this since we merged https://github.com/galaxyproject/galaxy/pull/9345

**Pull request closed without merging**
